### PR TITLE
GitHub Issue NOAA-EMC/GSI#298.  Convert Hera and Orion modulefiles to lua format

### LINKED ISSUE #298: Convert Hera/Orion modulefiles to lua for DA GFSv16.2.0
Work is underway to convert Hera and Orion modulefiles to lua format (please see global-workflow issue [#639](https://github.com/NOAA-EMC/global-workflow/issues/639)) for the WCOSS2 port of GFSv16.1.5 (GFSv16.2.0).

Two new modulefiles will be created:

modulefiles/modulefile.ProdGSI.hera.lua
modulefiles/modulefile.ProdGSI.orion.lua

and ush/build_all_cmake.sh will be updated to use the new modulefiles.

### PR DESCRIPTION
**Pull request by MichaelLueken**
Convert Hera and Orion modulefiles to lua format.  Reduce cmake_minimum_required from 3.16.2 to 3.16.1, in CMakeLists.txt files, to allow the code to compile on Hera.  Update ush/build_all_cmake.sh to compile using lua formatted modulefiles for Hera and Orion.  Update ush/sub_hera to allow the job submission driver to submit jobs using new lua formatted modulefile for Hera.

Resolves issue #298

### CODE REVIEW

**Comment by MichaelLueken-NOAA** [comment]
@RussTreadon-NOAA As per the notification made by Arun yesterday, I have converted the Hera and Orion modulefiles to lua format in feature/gfsda.v16.1.5_wcoss2_port (GFSv16.2.0).  I have been able to confirm that the updates work on Hera, but would like for you to try building on Orion.  Thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,7 +72,7 @@ project(GSI)
   option(BUILD_REG_TESTING "Build the Regression Testing Suite" ON)
   option(BUILD_UNIT_TESTING "Build the Unit Testing Suite" OFF)
   option(BUILD_GSDCLOUD_ARW "Build the GSD cloud analysis " OFF)
-  cmake_minimum_required(VERSION 3.16.2)
+  cmake_minimum_required(VERSION 3.16.1)
```
Comment (RussTreadon-NOAA) [inline CMakeLists.txt:75]: This is interesting.   Why do we need to decrease the cmake version when moving to lua format?

Comment (RussTreadon-NOAA) [reply CMakeLists.txt:75]: My bad.  Now I see what is happening.    Comment resolved.

**Review by RussTreadon-NOAA — COMMENTED** [review]
sub_hera has been updated.   Has your fork been installed on Hera, DA apps built, and regression tests run?

I built the forked gfsda.v16.1.5_wcoss2_port on Orion.   A C192L127 case was run in 3dvar mode.   The same case was run using a global_gsi.x built from the master.  Analysis results differ but this is expected since gfsda_v16.1.5_wcoss2_port does not contain all the updates found in the master.

A special note regarding CRTM_FIX.   The master modulefile.ProdGSI.orion defines CRTM_FIX because the hpc/1.1.0 crtm/2.3.0 does not define CRTM_FIX.    The forked modulefile.ProdGSI.orion.lua also loads hpc/1.1.0 crtm/2.3.0.   However, modulefile.ProdGSI.orion.lua does not define CRTM_FIX.   If developers assume CRTM_FIX is defined by crtm/2.3.0 or in modulefile.ProdGSI.orion.lua, they will encounter problems.   It's preferable to have CRTM_FIX defined by the CRTM module.

**Comment by MichaelLueken-NOAA** [comment]
@RussTreadon-NOAA Thank you for reviewing the changes.  I was able to build the code on Hera, but only some of the regression tests ran (since there are no fix files for the regression tests in release branches) and those that did run didn't reproduce the current master.

For the CRTM_FIX issue, on Hera, the hpc-stack crtm/2.3.0 module has had the CRTM_FIX defined:

[Michael.Lueken@hfe02 build]$ echo $CRTM_FIX
/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-18.0.5.274/crtm/2.3.0/fix

Is this not the case for Orion?  If it isn't, then I will need to create an hpc-stack issue and ask them to please install the CRTM coefficient files on Orion.

**Comment by RussTreadon-NOAA** [comment]
I did the following from a clone of your forked `feature/gfsda.v16.1.5_wcoss2_port` on Orion

```
module use /work/noaa/da/Russ.Treadon/git/gsi/pr299/modulefiles
module load modulefile.ProdGSI.orion
module list
```

I confirmed that `crtm/2.3.0` was loaded.  `module show crtm/2.3.0` returns

```
--------------------------------------------------------------------------------------------------------------------
   /apps/contrib/NCEP/libs/hpc-stack/modulefiles/compiler/intel/2018.4/crtm/2.3.0.lua:
--------------------------------------------------------------------------------------------------------------------
help([[]])
conflict("crtm")
setenv("crtm_ROOT","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.3.0")
setenv("crtm_VERSION","2.3.0")
setenv("CRTM_INC","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.3.0/include")
setenv("CRTM_LIB","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.3.0/lib/libcrtm.a")
whatis("Name: crtm")
whatis("Version: 2.3.0")
whatis("Category: library")
whatis("Description: crtm library")
```

`CRTM_FIX` is not defined by `crtm/2.3.0`.   Interestingly, `module show crtm/2.4.0` on Orion defines `CRTM_FIX`

```
Orion-login-1[21] rtreadon$ module show crtm/2.4.0
--------------------------------------------------------------------------------------------------------------------
   /apps/contrib/NCEP/libs/hpc-stack/modulefiles/compiler/intel/2018.4/crtm/2.4.0.lua:
--------------------------------------------------------------------------------------------------------------------
help([[]])
conflict("crtm")
setenv("crtm_ROOT","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.4.0")
setenv("crtm_VERSION","2.4.0")
setenv("CRTM_INC","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.4.0/include")
setenv("CRTM_LIB","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.4.0/lib/libcrtm.a")
setenv("CRTM_FIX","/apps/contrib/NCEP/libs/hpc-stack/intel-2018.4/crtm/2.4.0/fix")
whatis("Name: crtm")
whatis("Version: 2.4.0")
whatis("Category: library")
whatis("Description: crtm library")
```

I went through the same procedure on Hera.  As you note, `crtm/2.3.0` defines `CRTM_FIX`.  `crtm/2.3.0` defines 

`setenv("CRTM_FIX","/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-18.0.5.274/crtm/2.3.0/fix")`

whereas `crtm/2.4.0` on Hera defines

`setenv("CRTM_FIX","/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-18.0.5.274/crtm/2.4.0/fix")`

A diff of these two `CRTM_FIX` shows that they are not identical.  `crtm/2.4.0` contains more coefficient files.   `amsua_metop-c.SpcCoeff.bin` differs between the two versions.

**Review by RussTreadon-NOAA — APPROVED** [review]
Thanks, @MichaelLueken-NOAA, for contacting Hang.  It will be good to add `CRTM_FIX` to `crtm/2.3.0`.    Good to hear that the regression tests run on Hera since we will eventually merge `feature/gfsda.v16.1.5_wcoss2_port` into the `master`.

The changes in this PR seem to be ready to be merged into the authoritative `feature/gfsda.v16.1.5_wcoss2_port`

**Comment by MichaelLueken-NOAA** [comment]
Russ has approved the update (Thanks @RussTreadon-NOAA!), I will now merge this back to the NOAA-EMC/GSI feature/gfsda.v16.1.5_wcoss2_port branch.

**Pull request merged**